### ISSUE TITLE
[Feature Request] local address for server

### ISSUE THREAD

**Issue by Efreak** [CappielloAntonio/tempo#175]
Please add a local address for the server like dsub uses. When available, connect and use the local server; when not available, connect over the internet.

There's several ways to implement this:
- ping the local server and check if it's correct type of server in some way (detect from HTML?)
- use SSID (wifi network name) (dsub and other apps use this. Easier than above, more secure than below)
- just assume the server is correct and try to log in (possibly disallow insecure login if this method is chosen)

**Comment by cstby** [comment]
I'll add that this would be incredibly useful for those of us trying to be careful about server bandwidth or home Internet data usage. I would love to set a transcoding policy based on whether Tempo is connected over LAN or WAN.

**Comment by CappielloAntonio** [comment]
Hi, I have implemented the addition of the local address to which to connect instead of using the main address. The algorithm is something like this:
If connected to a local address, it continues to use this until it is reachable, after which it switches to the main address.

If connected to the main address, try to connect to the local address after a fixed X amount of time has passed. If the latter can be reached, use the local one, otherwise continue using the main one.

If the main one is also not reachable then I present a warning message about the unreachability of the server.

The check on which type of connection to use is done every time you enter the app, after a pre-established X time.

At the moment I don't think I will implement an option for ad hoc transcoding by type of address used since the transcoding option by type of connection used already exists. I think this type of control is "stronger" than the previous one.

You can reopen this issue or a new issue if you encounter any problems related to this implementation.

[app-notquitemy-debug.zip](https://github.com/user-attachments/files/15521992/app-notquitemy-debug.zip)

**Comment by CappielloAntonio** [comment]
The functionality still needs to be fully tested before we can release it, but if you prefer you can test it. 
I'm closing this issue, but feel free to reopen it or open a new one if you find any problems.

[app-notquitemy-debug.zip](https://github.com/user-attachments/files/15526304/app-notquitemy-debug.zip)
